ext/standard/quot_print.c

```diff
@@ -34,16 +34,12 @@ static char php_hex2int(int c) /* {{{ */
 {
 	if (isdigit(c)) {
 		return c - '0';
-	}
-	else if (c >= 'A' && c <= 'F') {
+	} else if (c >= 'A' && c <= 'F') {
 		return c - 'A' + 10;
-	}
-	else if (c >= 'a' && c <= 'f') {
+	} else {
+		ZEND_ASSERT(c >= 'a' && c <= 'f');
 		return c - 'a' + 10;
 	}
-	else {
-		return -1;
-	}
```

Comment:
I'm wondering if we should not just use a bitwise operation to make it unconditionally lower/uppercase so that we don't need to do another `if` check.

Reply:
That might be probably more expensive if I understand correct you want to use store (not just load). What you can do is is maybe to get rid of one condition but not sure if it's worth the readability:
```c
if (isdigit(c)) {
	return c - '0';
} else if (c >= 'a') {
	ZEND_ASSERT(c <= 'f');
	return c - 'a' + 10;
} else {
	ZEND_ASSERT(c >= 'A' && c <= 'F');
	return c - 'A' + 10;
}
```